src/node/stage/approved.rs

```diff
@@ -2025,6 +1975,10 @@ impl Approved {
             }
         }
 
+        // FIXME: this should not be necessary because elder membership should always be up to date
+        // after parsec reset. But without this almost all the tests fail. Figure out why.
+        self.members_changed = true;
```

Comment:
reset_parsec will be called for pruning as well. 
setting members_change to true for such case sounds bit weird?

Reply:
Yes, that's why I put in the FIXME. It's not that critical though, because the flag is there only to prevent calling `promote_and_demote_elders` too often, as it's quite expensive. So worst case is we might call it unnecessarily a few times, wasting some CPU cycles.